tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts

```diff
@@ -38,15 +41,39 @@ export const getApiVersionType: IApiVersionTypeExtractor = async (
 };
 
 export const isModelOnly: IModelOnlyChecker = async (packageRoot: string): Promise<boolean> => {
-    // For MLC, simply check for parameters.ts - its absence indicates a model-only service
+    // Check locally for parameters.ts
     const parametersPath = join(packageRoot, "src/rest/parameters.ts");
     const isParametersExists = await exists(parametersPath);
-    
+
     if (!isParametersExists) {
-        logger.warn(`No parameters.ts found in ${packageRoot}, this is a model-only service in Modular client`);
-        return true;
+        // Get npm view to find the latest stable version
+        const packageName = getNpmPackageName(packageRoot);
+        const npmViewResult = await tryGetNpmView(packageName);
+        if (!npmViewResult) {
+            logger.warn(`No npm package found for ${packageName}, cannot check GitHub directory`);
+            return true;
+        }
+
+        const stableVersion = getLatestStableVersion(npmViewResult);
+        if (!stableVersion) {
+            logger.warn(`No stable version found for ${packageName}, cannot check GitHub directory`);
+            return true;
+        }
+
+        // Check if src/api exists in the GitHub repository
+        const hasOperationsInGithub = await checkDirectoryExistsInGithub(
+            packageRoot,
+            "src/api",
+            packageName,
+            stableVersion
+        );
+
+        if (!hasOperationsInGithub) {
+            logger.warn(`No parameters.ts found in ${packageRoot} and no src/api directory found in GitHub for ${packageName}, this is a model-only service`);
+            return true;
+        }
```

Comment:
Since a condition has been added here, it can now cover the modular case and correctly determine whether it is modelOnly.